src/main/scala/wolfendale/nunjucks/TemplateParser.scala

```diff
@@ -51,16 +51,51 @@ object TemplateParser {
 
       import NoWhitespace._
       P(
-        (`if` ~ partial)
-          .map(TemplateNode.If.ConditionalContent.tupled) ~ (elif ~ partial)
-          .map(TemplateNode.If.ConditionalContent.tupled)
-          .rep ~ (`else` ~ partial).? ~ endIf)
+        (`if` ~ partial).map(TemplateNode.If.ConditionalContent.tupled)
+          ~ (elif ~ partial).map(TemplateNode.If.ConditionalContent.tupled).rep
+          ~ (`else` ~ partial).?
+          ~ endIf
+      )
         .map {
           case (x, xs, e) =>
             TemplateNode.If(x +: xs, e)
         }
     }
 
+    def switchTag = {
+      def switch = {
+        import SingleLineWhitespace._
+        P( openTag ~ "switch" ~/ Parser.expression ~ closeTag)
+      }
+
+      def `case` = {
+        import SingleLineWhitespace._
+        P( openTag ~ "case" ~/ Parser.expression ~ closeTag)
+      }
+
+      def `default` = {
+        import SingleLineWhitespace._
+        P( openTag ~ "default" ~ closeTag)
+      }
+
+      def endSwitch = {
+        import SingleLineWhitespace._
+        P( openTag ~ "endswitch" ~ closeTag)
+      }
+
+      import NoWhitespace._
+      P(
+        switch
+          ~ (`case` ~ partial).map(a => TemplateNode.Switch.ConditionalContent(Seq(a._1), a._2)).rep
```

Comment:
Can switch statements be empty?

Reply:
Looking at some examples it seems that switch statements can't be empty, but if there's a default there can be no cases.

```
{% switch foo %}
{% endswitch %}
```

does not compile, whereas:
```
{% switch foo %}
{% default %}
{% endswitch %}
```
does

Reply:
two tests added to demonstrate this